Search failed if deployed using shards caused by `collection not existed`

**Please state your issue using the following template and, most importantly, in English.**

**Describe the bug**
```
__________________ TestCompactBase.test_search_after_compact ___________________

self = <test_compact.TestCompactBase object at 0x7fcc76dba320>
connect = <milvus.client.stub.Milvus object at 0x7fccf4aff2b0>
collection = 'test_8jfSax6M'

    @pytest.mark.timeout(COMPACT_TIMEOUT)
    def test_search_after_compact(self, connect, collection):
        '''
        target: test search after compact
        method: after compact operation, search vector
        expected: status ok
        '''
        vectors = gen_vector(nb, dim)
        status, ids = connect.insert(collection, vectors)
        assert status.OK()
        status = connect.flush([collection])
        assert status.OK()
        status = connect.compact(collection)
        assert status.OK()
        query_vecs = [vectors[0]]
        status, res = connect.search(collection, top_k, query_records=query_vecs)
        logging.getLogger().info(res)
>       assert status.OK()
E       AssertionError: assert False
E        +  where False = <bound method Status.OK of Status(code=4, message='Collection test_8jfSax6M does not exist. Use milvus.has_collection to verify whether the collection exists. You also can check whether the collection name exists.')>()
E        +    where <bound method Status.OK of Status(code=4, message='Collection test_8jfSax6M does not exist. Use milvus.has_collection to verify whether the collection exists. You also can check whether the collection name exists.')> = Status(code=4, message='Collection test_8jfSax6M does not exist. Use milvus.has_collection to verify whether the collection exists. You also can check whether the collection name exists.').OK

test_compact.py:444: AssertionError
```


**Steps/Code to reproduce behavior**
http://internal.zilliz.com:18080/jenkins/job/milvus-ci/job/0.10.4/71/artifact/_artifacts/0-10-4/centos7/gpu/milvus_gpu_shards_dev_test.log
**Expected behavior**
A clear and concise description of what you expected to happen.

**Method of installation**
- [ ] Docker/cpu
- [ ] Docker/gpu
- [ ] Build from source

**Environment details**
- Hardware/Softwars conditions (OS, CPU, GPU, Memory)


- Milvus version (master or released version)


**Configuration file**
Settings you made in `server_config.yaml` or `milvus.yaml`

```yaml
paste-file-content-here
```

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,6 +3,7 @@
 Please mark all change in change log and use the issue from GitHub
 # Milvus 0.10.5 (TBD)
 ## Bug
+-   \#4296,#4554 Fix mishards add name-mismatched pod to read-only group
 -   \#4307 Specify partition to load for load_collection()
 -   \#4378 Multi-threads to call load_collection() and search() cause Milvus hang
 -   \#4484 Milvus only search default partition if search parameter 'partition_tags' contains '_default'

--- a/shards/discovery/plugins/kubernetes_provider.py
+++ b/shards/discovery/plugins/kubernetes_provider.py
@@ -208,6 +208,10 @@ class EventHandler(threading.Thread):
 
         pods_with_event = set()
         for each_event in event['events']:
+            if not re.match(self.pod_patt, event['pod']):
+                # logger.warning(f"Do not match pod {event['pod']}")
+                # Pod name mismatch, do not add it
+                return
             pods_with_event.add(each_event['pod'])
             if each_event['ready']:
                 self.record_pending_add(each_event['pod'],

--- a/shards/discovery/plugins/kubernetes_provider.py
+++ b/shards/discovery/plugins/kubernetes_provider.py
@@ -208,8 +208,8 @@ class EventHandler(threading.Thread):
 
         pods_with_event = set()
         for each_event in event['events']:
-            if not re.match(self.pod_patt, event['pod']):
-                # logger.warning(f"Do not match pod {event['pod']}")
+            if not re.match(self.pod_patt, each_event['pod']):
+                # logger.warning(f"Do not match pod {each_event['pod']}")
                 # Pod name mismatch, do not add it
                 return
             pods_with_event.add(each_event['pod'])

